design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,188 @@
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management.
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+The below user stories describe sample needs of a site administrator. Fabric administrators' requirements aren't elaborated.
+
+1. As a site administrator, I want to join my cluster with an existing fabric. After receiving the credentials (e.g., fabric certificate and site keypair, I would like to provide minimal additional configuration needed to deploy ClusterLink to the site/cluster I manage.
+1. As a site administrator, I would like to scale my ClusterLink deployment to meet growing demand (e.g., bandwidth) for cross-site communication.
+1. As a site administrator, I would like to completely remove ClusterLink from my cluster, restoring its previous state.
+
+## Goals
+
+This design document should:
+
+* Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+* Define the task functions of the ClusterLink deployment operator for setting up the ClusterLink components in the cluster.
+* Define the CRD (Custom Resource Definition) for the ClusterLink deployment operator.
+
+## Non-Goals
+
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+* This document focuses only on the deployment of ClusterLink to the K8s cluster.
+The deployment to another environment, such as VMs, is out of scope.
+* Security considerations related to the creation of certificates for the fabric or each site by the fabric administrator are also beyond the scope of this document.
+
+## Proposal
+
+The proposal describes how the ClusterLink project can be deployed on any K8s Cluster.
+In this proposal, we distinguish between two entities:
+
+1. Fabric administrator: manages the fabric and is responsible for determining which clusters can join the fabric.
+1. Site administrator: manages the k8s Cluster and is responsible for deploying the ClusterLink into the cluster.
+
+Before deploying the ClusterLink, a few prerequisite steps should be completed:
+
+1. The fabric administrator should create the site certificates (public and private) and transfer them, along with the public CA certificate, to the site administrator.
+2. The site administrator should deploy the certificates as k8s secrets to the cluster.  This step can be automated to reduce operator toll and chance of introducing errors.
+3. The site administrator should deploy the ClusterLink operator to the cluster and register ClusterLink Custom Resource Definition (CRD) class to the cluster. This step is not dependent on the previous two and can be done at the administrator's convenience.
+
+After completing all the prerequisites, the site administrator can edit and apply a ClusterLink CRD instance to the operator. The operator will then create the ClusterLink components, including:
+
+* A ClusterLink dedicated namespace, which by default will be cl-operator-ns.
```

Comment:
The operator namespace is required for running the operator itself and thus can't be something the operator creates.
Did you mean the namespace dedicated for ClusterLink components (e.g., `clusterlink-system`)?
If so, we may also want to require that it is created ahead of time and assume that CL always deploys to an existing namespace (this would also reduce the RBAC permissions needed by the operator). If we decide to accept that the namespace does not exist, we should create and also remove it (probably needs a `Finalizer` or at least a label for us to know when we create a namespace and when it pre-existed.